web/components/AccessManagement/DeletePopup.tsx

```diff
@@ -29,16 +31,31 @@ function DeletePopup(props: PropertyType) {
   } = props;
   const { isOpen, onOpen, onClose } = useDisclosure();
   const cancelRef = useRef(null);
+  const mutation = trpc.account.remove.useMutation();
+  const [showError, setShowError] = useState<boolean>(false);
+  const [errorMessage, setErrorMessage] = useState<string>("");
 
-  function handleDelete() {
-    var temp = accountList.filter(
+  const handleDelete = () => {
+    var newArr = accountList.filter(
       (e) => itemsToDelete.indexOf(accountList.indexOf(e)) < 0
     );
-
-    updateAccountList(temp);
-    updateSelectItems(false);
-    updateItemsToDelete([]);
-  }
+    var removeArr = accountList.filter(
+      (e) => itemsToDelete.indexOf(accountList.indexOf(e)) > -1
+    );
```

Comment:
Consider improving the readability for these functions. For `removeArr`, you can use `.filter` over accountList but pull the index from the method as follows `.filter((e, i) => {})`. Use the `.includes` method to see if the index `i` is in the array of indices that need to be removed.

For `newArr`, you can do something similar by negating the `.includes` inside of the filter.

There's a more efficient way of doing this since `.includes` is O(n) by itself, but it won't matter that much. We'll prioritize readability here.